src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -0,0 +1,42 @@
+import datetime
+
+from sentry.exceptions import InvalidConfiguration
+from sentry.utils import redis
+
+from . import base
+
+
+class RedisRealtimeMetricsStore(base.RealtimeMetricsStore):
+    def __init__(
+        self,
+        cluster: str,
+        counter_bucket_size: int,
+        counter_ttl: datetime.timedelta,
+        prefix: str,
+    ) -> None:
+        self.cluster = redis.redis_clusters.get(cluster)
+        self._counter_bucket_size = counter_bucket_size
+        self._counter_ttl = int(counter_ttl / datetime.timedelta(milliseconds=1))
+        self._prefix = prefix
+
+    def validate(self) -> None:
+        if self._counter_bucket_size <= 0:
+            raise InvalidConfiguration("bucket size must be at least 1")
+
+    def increment_project_event_counter(self, project_id: int, timestamp: float) -> None:
```

Comment:
`timestamp: Union[int, float]` maybe?  Ideally there'd be a `Timestamp` newtype or so... but as the stdlib doesn't introduce that yet perhaps there's no consensus on this so lets just go with a union for now